api/worker_grpc.go

```diff
@@ -686,6 +686,65 @@ func notifyWorkersPremieres(daoWrapper dao.DaoWrapper) {
 	}
 }
 
+type generateVideoSectionImagesParameters struct {
+	sections                                    []model.VideoSection
+	playlistUrl, courseName, courseTeachingTerm string
+	courseYear                                  uint32
+}
+
+func GenerateVideoSectionImages(daoWrapper dao.DaoWrapper, parameters *generateVideoSectionImagesParameters) error {
+	workers := daoWrapper.WorkerDao.GetAliveWorkers()
+	workerIndex := getWorkerWithLeastWorkload(workers)
+	workers[workerIndex].Workload += 1 // Not sure if that's a reasonable amount.
+	conn, err := dialIn(workers[workerIndex])
+	defer func() {
+		endConnection(conn)
+		workers[workerIndex].Workload -= 1
+	}()
+	if err != nil {
+		log.WithError(err).Error("Unable to dial server")
+		return err
+	}
+
+	client := pb.NewToWorkerClient(conn)
+
+	// remove old images
+	_, err = client.CleanSectionImageFolder(context.Background(), &pb.CleanSectionImageFolderRequest{
```

Comment:
Wouldn't it make sense to instead of delete everything only regenerate the files the worker doesn't already have on disk?

Reply:
Also I don't think this needs a extra endpoint, the worker can clean up the directory when it gets a `GenerateSectionImage` request